spinnman/processes/get_machine_process.py

```diff
@@ -68,24 +85,19 @@ def _make_chip(self, chip_info, machine):
         :rtype: :py:class:`spinn_machine.Chip`
         """
 
-        # Create the processor list
-        processors = list()
-        n_cores = min(chip_info.n_cores, Machine.MAX_CORES_PER_CHIP)
+        # Create the down cores set if any
+        n_cores = min(chip_info.n_cores, Machine.max_cores_per_chip())
         core_states = chip_info.core_states
-        for virtual_core_id in range(n_cores):
-            # Add the core provided it is not to be ignored
-            if ((chip_info.x, chip_info.y, virtual_core_id) not in
-                    self._ignore_cores):
-                if virtual_core_id == 0:
-                    processors.append(Processor.factory(
-                        virtual_core_id, is_monitor=True))
-                elif core_states[virtual_core_id] == CPUState.IDLE:
-                    processors.append(Processor.factory(virtual_core_id))
-                else:
-                    logger.warning(
-                        "Not using core {}, {}, {} in state {}",
-                        chip_info.x, chip_info.y, virtual_core_id,
-                        core_states[virtual_core_id])
+        down_cores = self._ignore_cores_map.get(
+            (chip_info.x, chip_info.y))
+        for i in range(1, n_cores):
+            if core_states[i] != CPUState.IDLE:
+                logger.warning(
+                    "Not using core {}, {}, {} in state {}",
+                    chip_info.x, chip_info.y, i,  core_states[i])
+                if down_cores is None:
```

Comment:
ignore cores map is a default dict of a set. so itll never reutrn None.

Reply:
Wrong It will normally be none!

Why you ask.
Because the call is 
self._ignore_cores_map.get((chip_info))
and not
self._ignore_cores_map[chip_info]

see https://docs.python.org/2/library/collections.html#collections.defaultdict

Reply:
Name get clearer to show it has the default None